tests/unit/Framework/ExecutionOrderDependencyTest.php

```diff
@@ -0,0 +1,188 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+use PHPUnit\Framework\ExecutionOrderDependency;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \PHPUnit\Framework\ExecutionOrderDependency
+ */
+class ExecutionOrderDependencyTest extends TestCase
+{
+    public static function createFromParametersProvider(): array
+    {
+        return [
+            // Dependency on specific class::method target
+            ['Class1', 'test1', 'Class1::test1', false],
+            ['Class2', 'test2', 'Class2::test2', false],
+            ['Class3::method3', null, 'Class3::method3', false],
+
+            // Dependency on whole class
+            ['Class4', null, 'Class4::class', true],
+            ['Class5', '', 'Class5::class', true],
+            ['Class6', 'class', 'Class6::class', true],
+            ['Class7::class', null, 'Class7::class', true],
+        ];
+    }
+
+    public static function createWithCloneOptionProvider(): array
+    {
+        return [
+            // Cloning option values
+            ['clone', false, true],
+            ['!clone', false, false],
+            ['shallowClone', true, false],
+            ['!shallowClone', false, false],
+        ];
+    }
+
+    public function testCreateDependencyOnClassFromClassNameOnly(): void
+    {
+        $dependency = new ExecutionOrderDependency('ClassDependency');
+
+        $this->assertTrue($dependency->targetIsClass());
+        $this->assertSame('ClassDependency::class', $dependency->getTarget());
+        $this->assertSame('ClassDependency', $dependency->getTargetClassName());
+    }
+
+    public function testDependsAnnotationRequireATargetToBeValid(): void
+    {
+        $dependency = ExecutionOrderDependency::createFromDependsAnnotation('SomeClass', '');
+        $this->assertFalse($dependency->isValid());
```

Comment:
```suggestion
        $dependency = ExecutionOrderDependency::createFromDependsAnnotation('SomeClass', '');

        $this->assertFalse($dependency->isValid());
```

Reply:
we are not paid by the line @localheinz ;-)